src/schemas/settings/index.ts

```diff
@@ -0,0 +1,22 @@
+import { z } from 'zod'
+import * as v1 from './v1'
+import * as v2 from './v2'
+import { exhaustive } from '../../utils/typescript'
+
+const AnySettingSchema = z.discriminatedUnion('version', [
+  v1.AppSettingsSchema,
+  v2.AppSettingsSchema,
+])
+
+export function migrate(settings: z.infer<typeof AnySettingSchema>) {
+  switch (settings.version) {
+    case '1.0':
+      return migrate(v1.migrate(settings))
+    case '2.0':
+      return settings
```

Comment:
This function will run recursively until the latest version of the schema is reached.